tests/linting/Sniffs/Files/LineLengthSniff.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+class Linting_Sniffs_Files_LineLengthSniff extends Generic_Sniffs_Files_LineLengthSniff
+{
+    /**
+     * Notes the types of tokens which are comments so their lines can be skipped.
+     *
+     * @var array
+     */
+    CONST COMMENT_TOKENS = [
+        T_DOC_COMMENT,
+        T_DOC_COMMENT_STAR,
+        T_DOC_COMMENT_WHITESPACE,
+        T_DOC_COMMENT_TAG,
+        T_DOC_COMMENT_OPEN_TAG,
+        T_DOC_COMMENT_CLOSE_TAG,
+        T_DOC_COMMENT_STRING,
+    ];
+    /**
+     * The limit that the length of a line should not exceed.
+     *
+     * @var int
+     */
+    public $lineLimit = 120;
+
+    /**
+     * The limit that the length of a line must not exceed.
+     *
+     * Set to zero (0) to disable.
+     *
+     * @var int
+     */
+    public $absoluteLineLimit = 0;
+
+    /**
+     * @inheritDoc
+     */
+    public function process(PHP_CodeSniffer_File $phpcsFile, $stackPtr)
+    {
+        $tokens = $phpcsFile->getTokens();
+        for ($i = 1; $i < $phpcsFile->numTokens; $i++) {
+            if ($tokens[$i]['column'] === 1) {
+                $lineIsComment = false;
+                $j = $i;
+                do {
+                    $lineIsComment = $lineIsComment || in_array($tokens[$j]['code'], self::COMMENT_TOKENS);
```

Comment:
When we find a line, see whether any of its elements is one of the comment-type elements.